app/controllers/BulkFileUploadController.scala

```diff
@@ -96,35 +109,54 @@ class BulkFileUploadController @Inject()(
     } yield (SessionState(u.flatMap(_.fileUploadState), u))
   }
 
+  // GET /file-rejected
+  final def markFileUploadAsRejected: Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
+
+    UpscanUploadErrorForm.bindFromRequest().fold(
+      _ => Future.successful(BadRequest),
+      s3Error =>
+        sessionState(request.internalId).flatMap { ss =>
+          ss.state match {
+            case Some(s) => fileUploadWasRejected(s3Error)(s).flatMap { newState =>
+              updateSession(newState, ss.userAnswers).map { res =>
+                Redirect(routes.BulkFileUploadController.showFileUpload())
+              }
+            }
+            case None => Future.successful(InternalServerError("Missing file upload state"))
+          }
+        }
+    )
+  }
+
   private def updateSession(newState: FileUploadState, userAnswers: Option[UserAnswers]) = {
     if (userAnswers.nonEmpty)
       sessionRepository.set(userAnswers = userAnswers.get.copy(fileUploadState = Some(newState)))
     else Future.successful(true)
   }
 
-  final def successRedirect(id: String, userAnswers: UserAnswers)(implicit rh: RequestHeader) = appConfig.baseExternalCallbackUrl + (rh.cookies.get(COOKIE_JSENABLED) match {
-    case Some(_) => getBulkEntryDetails(userAnswers)
-    case None => controller.showWaitingForFileVerification
+  final def successRedirect(id: String)(implicit rh: RequestHeader) = appConfig.baseExternalCallbackUrl + (rh.cookies.get(COOKIE_JSENABLED) match {
```

Comment:
This needs refactoring. Get rid of this method and in the callback add the url directly